yarn-project/archiver/src/archiver/kv_archiver_store/block_store.ts

```diff
@@ -73,6 +77,39 @@ export class BlockStore {
     });
   }
 
+  /**
+   * Unwinds blocks from the database
+   * @param from -  The tip of the chain, passed for verification purposes,
+   *                ensuring that we don't end up deleting something we did not intend
+   * @param blocksToUnwind - The number of blocks we are to unwind
+   * @returns True if the operation is successful
+   */
+  unwindBlocks(from: number, blocksToUnwind: number) {
+    return this.db.transaction(() => {
+      // We should only allow deleting the very last ye, otherwise we can really get some messy shit.
+      const last = this.getSynchedL2BlockNumber();
+      if (from != last) {
+        throw new Error(`Can only remove the tip`);
+      }
+
+      for (let i = 0; i < blocksToUnwind; i++) {
+        const blockNumber = from - i;
+        const block = this.getBlock(blockNumber);
+
+        if (block === undefined) {
+          throw new Error(`Cannot remove block ${blockNumber} from the store, we don't have it`);
+        }
+        void this.#blocks.delete(block.data.number);
```

Comment:
Should we delete all the logs/contracts etc associated with the unwound blocks?

Reply:
Yes. I forgot, different store.

Reply:
Since all of these are added per block, seems like the apis could be made more convenient if we just have an `addBlock` and `unwindBlock` on the archiver_store and handle the logic in there. The archiver don't really need to know about these, as long as the store does.